src/PlutusTx/LedgerTypes.hs

```diff
@@ -0,0 +1,127 @@
+import Plutus.V1.Ledger.Contexts (ScriptContext,ScriptPurpose,TxInfo,TxInInfo)
+import Plutus.V1.Ledger.Credential (StakingCredential, Credential)
+import Plutus.V1.Ledger.Crypto (PubKey,PubKeyHash,PrivateKey,Signature)
+import Plutus.V1.Ledger.DCert (DCert)
+import Plutus.V1.Ledger.Interval (Interval, UpperBound, LowerBound, Extended)
+import Plutus.V1.Ledger.Value (CurrencySymbol, AssetClass, Value, TokenName)
+import Plutus.V1.Ledger.TxId (TxId)
+import Plutus.V1.Ledger.Tx (TxOut,TxOutRef)
+import Plutus.V1.Ledger.Time (POSIXTime,DiffMilliSeconds)
+import Plutus.V1.Ledger.Slot (Slot)
+import Plutus.V1.Ledger.Scripts (
+  Redeemer,
+  Datum,
+  ScriptHash,
+  ValidatorHash,
+  DatumHash,
+  MintingPolicyHash,
+  StakeValidatorHash,
+ )
+
+-- other imports
+import Data.Text
+
+writeLedgerTypes :: FilePath -> IO ()
+writeLedgerTypes fp = writeLedgerTypesAnd fp []
+
+writeLedgerTypesAnd :: FilePath -> [SumType 'Haskell] -> IO ()
+writeLedgerTypesAnd fp myTypes =
+  writePSTypes
+    fp
+    (buildBridge (defaultBridge <|> plutusBridge))
+    (ledgerTypes <> myTypes)
+
+ledgerTypes :: [SumType 'Haskell]
+ledgerTypes =
+  [ -- Newtypes or single constructor data types (extremelyUnsafeMkSumType is safe)
+    extremelyUnsafeMkSumType @Value
+  , extremelyUnsafeMkSumType @CurrencySymbol
+  , extremelyUnsafeMkSumType @AssetClass
+  , extremelyUnsafeMkSumType @TokenName
+  , extremelyUnsafeMkSumType @TxId
+  , extremelyUnsafeMkSumType @TxOut
+  , extremelyUnsafeMkSumType @TxOutRef
+  , extremelyUnsafeMkSumType @DiffMilliSeconds
+  , extremelyUnsafeMkSumType @POSIXTime
+  , extremelyUnsafeMkSumType @Slot
+  , extremelyUnsafeMkSumType @Redeemer
+  , extremelyUnsafeMkSumType @Datum
+  , extremelyUnsafeMkSumType @ScriptHash
+  , extremelyUnsafeMkSumType @ValidatorHash
+  , extremelyUnsafeMkSumType @DatumHash
+  , extremelyUnsafeMkSumType @MintingPolicyHash
+  , extremelyUnsafeMkSumType @StakeValidatorHash
+  , extremelyUnsafeMkSumType @PubKey
+  , extremelyUnsafeMkSumType @PubKeyHash
+  , extremelyUnsafeMkSumType @PrivateKey
+  , extremelyUnsafeMkSumType @Signature
+  , extremelyUnsafeMkSumType @TxInfo
+  , extremelyUnsafeMkSumType @TxInInfo
+  , extremelyUnsafeMkSumType @ScriptContext
+  , extremelyUnsafeMkSumType @LedgerBytes
+  , extremelyUnsafeMkSumType @Address
+  , extremelyUnsafeMkSumType @Ada
+  , extremelyUnsafeMkSumType @(Interval A)
+  , extremelyUnsafeMkSumType @(LowerBound A)
+  , extremelyUnsafeMkSumType @(UpperBound A)
+
+  -- True sum types, HasConstrIndices instances generated in the PlutusTx.ConstrIndices module
+  , mkSumTypeIndexed @DCert
+  , mkSumTypeIndexed @(Extended A)
+  , mkSumTypeIndexed @StakingCredential
+  , mkSumTypeIndexed @Credential
+  , mkSumTypeIndexed @ScriptPurpose
+  ]
+
+-- I'm leaving this commented b/c I'm not sure what the module structure for the ledger types should be.
+-- My assumption was that, like Plutarch, we'd just shove everything into it's respective Plutus.V1.Ledger module
```

Comment:
I agree, however, there's a slight complication in this story. CTL has its own Value type (quite a neat one) that they native support and work with.

What does that mean?

We'll if any of our Datum and Redeemers are using Value, it would make sense to map it to CTL Value and than have  `toLegacyValue :: CTL.Value -> Plutus.V1.Ledger.Value.Value` and ``fromLegacyValue :: Plutus.V1.Ledger.Value.Value -> CTL.Value ` used  in the ToData/FromData instances for `CTL.Value`

Does that make sense? Well anyway we need to iron this out.

Reply:
That also means that CTL would have to add this repo as a dependency, or we should have a Purescript module that does that conversion. Not sure what makes sense.

Reply:
Or Maybe we should have a little Purescript module here that takes care of such conversions?

Reply:
CTL should depend on us, I think. I ironed out the compatibility issues in the most recent PR here but I think it's a little confusing that we have to generate all the types except the three (?) that CTL has. 

Or maybe it's not that confusing and we can just add a note, I dunno. You're probably in a better position to decide since you've worked on CTL directly.

Reply:
Shouldn't CTL export all the ledger types? Like, isn't almost everyone who will use that library going to want those as well?